### PR TITLE
feat: integrate fps as submodule

### PR DESCRIPTION
**Pull request by prat-gpt**
## Description

Install the latest FPS as a submodule and integrate it for all the proposals

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

-   [X] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
-   [ ] ~added `!` to the type prefix if API or client breaking change~
-   [ ] ~provided a link to the relevant issue or specification~
-   [x] included the necessary unit and integration tests
-   [x] code is adequately commented
-   [ ] updated the relevant documentation or specification
-   [x] reviewed "Files changed" and left comments if necessary
-   [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

-   [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
-   [ ] ~confirmed `!` in the type prefix if API or client breaking change~
-   [x] confirmed all author checklist items have been addressed
-   [x] reviewed service design and naming
-   [ ] ~reviewed documentation is accurate~
-   [x] reviewed tests and test coverage
-   [x] manually tested (if applicable)

### CODE REVIEW

File: proposals/zips/zipTest.sol [hunk 1]
```diff
@@ -1,39 +1,24 @@
 //SPDX-License-Identifier: GPL-3.0-or-later
-pragma solidity 0.8.18;
+pragma solidity ^0.8.18;
 
 import {console} from "@forge-std/console.sol";
```
Comment (ElliotFriedman) [inline proposals/zips/zipTest.sol]: is this import needed?

File: test/integration/governance/GovernorDAO.sol [hunk 2]
```diff
@@ -6,9 +6,9 @@ import {Test} from "@forge-std/Test.sol";
 import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
 import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
 
-import {Addresses} from "@proposals/Addresses.sol";
-import {TestProposals} from "@proposals/TestProposals.sol";
-import {BaseTest} from "@test/integration/BaseTest.sol";
+import {Addresses} from "@forge-proposal-simulator/addresses/Addresses.sol";
+import {TestProposals} from "proposals/TestProposals.sol";
+import {BaseTest} from "test//integration/BaseTest.sol";
```
Comment (ElliotFriedman) [inline test/integration/governance/GovernorDAO.sol]: extra `/`

File: test/unit/finance/ERC20HoldingDeposit.t.sol [hunk 3]
```diff
@@ -3,12 +3,12 @@ pragma solidity 0.8.18;
 import "@forge-std/Test.sol";
 
 import {Core} from "@protocol/core/Core.sol";
-import {MockERC20} from "@test/mock/MockERC20.sol";
+import {MockERC20} from "test//mock/MockERC20.sol";
```
Comment (ElliotFriedman) [inline test/unit/finance/ERC20HoldingDeposit.t.sol]: extra `/`

File: test/unit/core/GlobalReentrancyLock.t.sol [hunk 4]
```diff
@@ -5,11 +5,11 @@ import {Address} from "@openzeppelin/contracts/utils/Address.sol";
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 import {Core} from "@protocol/core/Core.sol";
-import {getCore} from "@test/fixtures/Fixtures.sol";
+import {getCore} from "test//fixtures/Fixtures.sol";
```
Comment (ElliotFriedman) [inline test/unit/core/GlobalReentrancyLock.t.sol]: extra `/`

File: test/unit/finance/ERC20Splitter.t.sol [hunk 5]
```diff
@@ -4,12 +4,12 @@ import "@forge-std/Test.sol";
 
 import {Core} from "@protocol/core/Core.sol";
 import {Roles} from "@protocol/core/Roles.sol";
-import {MockERC20} from "@test/mock/MockERC20.sol";
+import {MockERC20} from "test//mock/MockERC20.sol";
```
Comment (ElliotFriedman) [inline test/unit/finance/ERC20Splitter.t.sol]: extra `/`

File: test/unit/finance/ERC20Splitter.t.sol [hunk 6]
```diff
@@ -4,12 +4,12 @@ import "@forge-std/Test.sol";
...
 import {ERC20Splitter} from "@protocol/finance/ERC20Splitter.sol";
-import {MockERC20, IERC20} from "@test/mock/MockERC20.sol";
+import {MockERC20, IERC20} from "test//mock/MockERC20.sol";
 
-import {getCore} from "@test/fixtures/Fixtures.sol";
-import {TestAddresses as addresses} from "@test/fixtures/TestAddresses.sol";
+import {getCore} from "test//fixtures/Fixtures.sol";
+import {TestAddresses as addresses} from "test//fixtures/TestAddresses.sol";
```
Comment (ElliotFriedman) [inline test/unit/finance/ERC20Splitter.t.sol]: extra `/`

File: test/unit/governance/GovernorDAO.t.sol [hunk 7]
```diff
@@ -9,11 +9,11 @@ import {GovernorCountingSimple} from "@openzeppelin/contracts/governance/extensi
 import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
 
 import {Core} from "@protocol/core/Core.sol";
-import {getCore} from "@test/fixtures/Fixtures.sol";
+import {getCore} from "test//fixtures/Fixtures.sol";
 import {Roles} from "@protocol/core/Roles.sol";
 import {GovernorDAO} from "@protocol/governance/GovernorDAO.sol";
-import {TestAddresses as addresses} from "@test/fixtures/TestAddresses.sol";
-import {MockERC20} from "@test/mock/MockERC20.sol";
+import {TestAddresses as addresses} from "test//fixtures/TestAddresses.sol";
```
Comment (ElliotFriedman) [inline test/unit/governance/GovernorDAO.t.sol]: extra `/`

File: test/unit/governance/GovernorDAO.t.sol [hunk 8]
```diff
@@ -9,11 +9,11 @@ import {GovernorCountingSimple} from "@openzeppelin/contracts/governance/extensi
...
+import {MockERC20} from "test//mock/MockERC20.sol";
```
Comment (ElliotFriedman) [inline test/unit/governance/GovernorDAO.t.sol]: extra `/`

File: test/unit/nfts/ERC1155AdminMinter.t.sol [hunk 9]
```diff
@@ -4,10 +4,10 @@ import "@forge-std/Test.sol";
 
 import {Core} from "@protocol/core/Core.sol";
 import {Roles} from "@protocol/core/Roles.sol";
-import {BaseTest} from "@test/BaseTest.sol";
+import {BaseTest} from "test//BaseTest.sol";
 import {ERC1155AdminMinter} from "@protocol/nfts/ERC1155AdminMinter.sol";
-import {TestAddresses as addresses} from "@test/fixtures/TestAddresses.sol";
-import {getSystem, configureSale, setSupplyCap} from "@test/fixtures/Fixtures.sol";
+import {TestAddresses as addresses} from "test//fixtures/TestAddresses.sol";
+import {getSystem, configureSale, setSupplyCap} from "test//fixtures/Fixtures.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/ERC1155AdminMinter.t.sol]: extra `/`

File: test/unit/nfts/ERC1155AutoGraphMinter.t.sol [hunk 10]
```diff
@@ -4,16 +4,16 @@ import "@forge-std/Test.sol";
...
+import {MockERC20, IERC20} from "test//mock/MockERC20.sol";
 import {GlobalReentrancyLock} from "@protocol/core/GlobalReentrancyLock.sol";
 import {ERC1155MaxSupplyMintable} from "@protocol/nfts/ERC1155MaxSupplyMintable.sol";
 import {ERC1155AutoGraphMinter} from "@protocol/nfts/ERC1155AutoGraphMinter.sol";
-import {TestAddresses as addresses} from "@test/fixtures/TestAddresses.sol";
-import {ERC1155AutoGraphMinterHelperLib as Helper} from "@test/helpers/ERC1155AutoGraphMinterHelper.sol";
-import {BaseTest} from "@test/BaseTest.sol";
+import {TestAddresses as addresses} from "test//fixtures/TestAddresses.sol";
+import {ERC1155AutoGraphMinterHelperLib as Helper} from "test//helpers/ERC1155AutoGraphMinterHelper.sol";
+import {BaseTest} from "test//BaseTest.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/ERC1155AutoGraphMinter.t.sol]: extra `/`

File: test/unit/nfts/GenesisHome.t.sol [hunk 11]
```diff
@@ -1,6 +1,6 @@
 pragma solidity 0.8.18;
 
-import {BaseTest} from "@test/BaseTest.sol";
+import {BaseTest} from "test//BaseTest.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/GenesisHome.t.sol]: extra `/`

File: test/unit/nfts/PlayTest.t.sol [hunk 12]
```diff
@@ -1,6 +1,6 @@
 pragma solidity 0.8.18;
 
-import {BaseTest} from "@test/BaseTest.sol";
+import {BaseTest} from "test//BaseTest.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/PlayTest.t.sol]: extra `/`

File: test/unit/nfts/seasons/ERC1155SeaonTwo.t.sol [hunk 13]
```diff
@@ -1,5 +1,5 @@
 pragma solidity 0.8.18;
 
-import {SeasonBase} from "@test/unit/nfts/seasons/SeasonBase.t.sol";
+import {SeasonBase} from "test//unit/nfts/seasons/SeasonBase.t.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/seasons/ERC1155SeaonTwo.t.sol]: extra `/`

File: test/unit/nfts/seasons/ERC1155SeasonOne.t.sol [hunk 14]
```diff
@@ -8,12 +8,12 @@ import {ERC1155MaxSupplyMintable} from "@protocol/nfts/ERC1155MaxSupplyMintable.
 
 import {ERC1155SeasonOne} from "@protocol/nfts/seasons/ERC1155SeasonOne.sol";
 import {TokenIdRewardAmount} from "@protocol/nfts/seasons/SeasonsBase.sol";
-import {TestAddresses as addresses} from "@test/fixtures/TestAddresses.sol";
+import {TestAddresses as addresses} from "test//fixtures/TestAddresses.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/seasons/ERC1155SeasonOne.t.sol]: extra `/`

File: test/unit/nfts/seasons/ERC1155SeasonOne.t.sol [hunk 15]
```diff
@@ -8,12 +8,12 @@ import {ERC1155MaxSupplyMintable} from "@protocol/nfts/ERC1155MaxSupplyMintable.
...
 import {ERC1155MaxSupplyMintable} from "@protocol/nfts/ERC1155MaxSupplyMintable.sol";
 
-import {SeasonBase} from "@test/unit/nfts/seasons/SeasonBase.t.sol";
+import {SeasonBase} from "test//unit/nfts/seasons/SeasonBase.t.sol";
 import {SeasonsTokenIdRegistry} from "@protocol/nfts/seasons/SeasonsTokenIdRegistry.sol";
-import {ERC1155SeaonsHelperLib as Helper} from "@test/helpers/ERC1155SeasonsHelper.sol";
+import {ERC1155SeaonsHelperLib as Helper} from "test//helpers/ERC1155SeasonsHelper.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/seasons/ERC1155SeasonOne.t.sol]: extra `/`

File: test/unit/nfts/seasons/SeasonBase.t.sol [hunk 16]
```diff
@@ -1,6 +1,6 @@
 pragma solidity 0.8.18;
 
-import {BaseTest} from "@test/BaseTest.sol";
+import {BaseTest} from "test//BaseTest.sol";
```
Comment (ElliotFriedman) [inline test/unit/nfts/seasons/SeasonBase.t.sol]: extra `/`

**Review by ElliotFriedman — COMMENTED** [review]
All minor nits, other than that

LGTM and LFG!

**Review by ElliotFriedman — APPROVED** [review]
LGTM!

**Review by ghost — APPROVED** [review]
Nice work @prateek105.

**Pull request merged**